api/server.js

```diff
@@ -0,0 +1,140 @@
+require('dotenv').config();
+
+const express = require('express');
+const helmet = require('helmet');
+const knex = require('knex');
+const bcrypt = require('bcryptjs'); // added
+const jwt = require('jsonwebtoken');
+const knexConfig = require('../knexfile.js');
+
+const server = express();
+
+const db = knex(knexConfig.development);
+
+server.use(helmet());
+server.use(express.json());
+
+server.get('/', (req, res) => {
+  res.send('hopefully everyone is up in here, up in here.');
+});
+
+server.post('/register', (req, res) => {
+  const userInfo = req.body;
+  // userinfo consists of username, name, and password
+  const hash = bcrypt.hashSync(userInfo.password, 12);
+
+  userInfo.password = hash;
+
+  db('users')
+    .insert(userInfo)
+    .then(ids => {
+      res.status(201).json(ids);
+    })
+    .catch(err => res.status(500).json(err));
+});
+
+///////=============== Middleware ===============///////
+// might want to clean up the routes and put these functions into the dbHelpers in the future
+
+function makeToken(user) {
+  //// i wonder if this will conflict with the above information!
+  const payload = {
+    username: user.username,
+    name: user.name,
+    department: [
+      'Department of Merrymaking',
+      'Ministry of Truth',
+      'Ministry of Thought'
+    ]
+  };
+  const secret = process.env.JWT_SECRET;
+
+  const options = {
+    expiresIn: '1000m'
+  };
+  return jwt.sign(payload, secret, options);
+}
+
+function checkDepartment(department) {
+  return function(req, res, next) {
+    if (req.decodedToken.department.includes(department)) {
+      next();
+    } else {
+      res.status(403).json({
+        message: `sorry yo, you need to be in ${department} to access this site.`
+      });
+    }
+  };
+}
+
+server.post('/login', (req, res) => {
+  const creds = req.body;
+  console.log(creds);
+
+  db('users')
+    .where({ username: creds.username })
+    .first()
+    .then(user => {
+      if (user && bcrypt.compareSync(creds.password, user.password)) {
+        const token = makeToken(user);
+        res.status(200).json({ message: `welcome ${user.name}`, token });
```

Comment:
You could put all of your numbers in a status object like `const statusObject = { success: 200 }`, then use the object like `res.status(statusObject.success)` to avoid 'magic numbers'.